### PR TITLE
Completed Edit Event

### PR DESCRIPTION
**Pull request by vwinstea**
The edit event page is complete now and is fully connected to the backend. I had to edit the routes a bit because I wanted to lift the state from Schedule.jsx so it could be passed to CreateEvent.jsx. I might try to connect the Create Event/Edit Event page with redux later this week, but I haven't worked with redux before so I'm not sure how long that will take.

### CODE REVIEW

File: src/components/pages/CreateShift.jsx [hunk 1]
```diff
@@ -139,10 +151,20 @@ export default function CreateShift() {
           <Form.Row>
             <Col><CancelButton onClick={() => history.push('/schedule')}>Cancel</CancelButton></Col>
             <Col xs={1} />
-            <Col><SubmitButton type="submit">Add Shift</SubmitButton></Col>
+            <Col><SubmitButton type="submit">{isEditing ? 'Save Edits' : 'Add Shift'}</SubmitButton></Col>
```
Comment (ctperry0301) [inline src/components/pages/CreateShift.jsx:154]: Could you change the text Save Edits to Update Shift?

**Review by ctperry0301 — APPROVED** [review]
I left one tiny comment asking you to change the text on a button, but after that this code looks good to be merged in. 

In general, there are a few different ways you could have gone about editing a form. I think the best way to do this would be to just pass route params into the CreateShift page. The next best way to do it would be to do what you did. (The difference is the first method is easier to follow, and also probably easier to implement).

Saying that, what you have now looks and works great. Since It exceeds the acceptance criteria and the code itself looks good/correct, you wont need to make any changes to how it works and is structured. 

Also, I wouldn't worry about adding redux to this yet, as redux hasn't been fully implemented to this project yet, adding it just to this component would require you to do a lot of the same work that I will be doing this week.

**Comment by vwinstea** [comment]
Sounds good, I just updated the button text so everything should be good to go now.

When you say to use route params, do you mean to pass the vigil id as a param and then fetch it from the backend? I considered that but I wasn't sure if it was good practice since we already have all of the data we need on the frontend. Where do you normally draw the line between necessary & unnecessary backend calls?

**Comment by ctperry0301** [comment]
> Sounds good, I just updated the button text so everything should be good to go now.
> 
> When you say to use route params, do you mean to pass the vigil id as a param and then fetch it from the backend? I considered that but I wasn't sure if it was good practice since we already have all of the data we need on the frontend. Where do you normally draw the line between necessary & unnecessary backend calls?

I just looked it up and the route params are exclusively in react native (for mobile apps) & they do not exist like I thought in regular React. So it turns out you can ignore what I said about using route params. Sorry for the confusion. 

In regards to where I would draw the line between necessary & unnecessary backend calls. I would say that it an ideal world, we would only call the backend when we are updating/adding something in the DB, or the first time we need to read something from our app. We typically don't want to read the same information from the backend more than once per session.

**Pull request merged**